packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -190,19 +231,21 @@ export class AwsCustomResource extends cdk.Construct implements iam.IGrantable {
     });
     this.grantPrincipal = provider.grantPrincipal;
 
-    if (props.policyStatements) {
-      for (const statement of props.policyStatements) {
-        provider.addToRolePolicy(statement);
-      }
-    } else { // Derive statements from AWS SDK calls
+    if (props.policy.resources) {
```

Comment:
Retain the original branch order to avoid unnecessary diffs.

Reply:
Will do